### ISSUE TITLE
add mqtt autodiscovery for homeassistant

### ISSUE THREAD

**Issue by pmagyar** [hcpy2-0/hcpy#52]
https://www.home-assistant.io/integrations/mqtt/#discovery-messages
we can create sensors directly in mqtt which the mqtt integration will automatically add to homeassistant.

Eg.
```
{
  "name": "Dishwasher",
  "unique_id": "dishwasher_SERIAL",
  "state_topic": "homeassistant/dishwasher/state",
  "value_template": "{{ value_json.PowerState }}",
  "payload_on": "On",
  "payload_off": "Off",
  "json_attributes_topic": "homeassistant/dishwasher/state",
  "device": {
    "identifiers": "dishwasher_SERIAL",
    "name": "Dishwasher",
    "manufacturer": "SIEMENS",
    "model": "SN55TS00CE",
    "serial_number": "SERIAL"
  },
  "availability": [
    {
      "topic": "homeassistant/LWT"
    },
    {
      "topic": "homeassistant/dishwasher/LWT"
    }
  ],
  "availability_mode": "latest"
}
```

<img width="931" alt="image" src="https://github.com/hcpy2-0/hcpy/assets/58977600/26dc3c91-ec26-49cd-9b8e-5deb0f3935df">

**Comment by Meatballs1** [comment]
What approach do you want to take? Do we want a library of known models with preformed configuration, or do you want to inspect the capabilities from `devices.json` and build a configuration from that?

**Comment by pmagyar** [comment]
I think generally we should use the `devices.json` as base.
To maintain a library with known models and expand that, doesn't sound sustainable. 
Maybe we need special configurations for some devices. This can be done through a library. 
But as start the `devices.json`  is enough.

**Comment by zibous** [comment]
I use the mqtt sensor setting (simple):

Example:

```yaml
mqtt:
    - sensor:
        - name: "Dishwasher"
          unique_id: "dishwasher_SERIAL"
          icon: mdi:dishwasher
          state_topic: "homeconnect/dishwasher/state"
          device:
            identifiers: "dishwasher_SERIAL"
            name: "Dishwasher"
            manufacturer: "SIEMENS"
            model: "SN55TS00CE"
            serial_number: "SERIAL"
            suggested_area: Kitchen
          availability:
            - topic: "homeconnect/dishwasher/LWT"
          availability_mode: all
        - name: "Internal Errot"
          unique_id: "dishwasher_InternalError"
          icon: mdi:dishwasher-alert
          state_topic: "homeconnect/dishwasher/state"
          value_template: "{{value_json.InternalError|default('Off')}}"
          payload_on: "On"
          payload_off: "Off"
          entity_category: diagnostic
          device:
            name: "Dishwasher"
            identifiers: "dishwasher_SERIAL"
          availability:
            - topic: "homeconnect/dishwasher/LWT"
          availability_mode: all
```
see: https://github.com/zibous/homeconnect-hcpy/blob/main/homeassistant/dishwasher.yaml

**Comment by jamesremuscat** [comment]
I started to play around with this before I discovered this issue: https://github.com/jamesremuscat/hcpy/tree/feature/ha-discovery

Currently generates discovery messages based on `devices.json`. My next step will be to try and work out how to specify appropriate Home Assistant "component types" - right now everything's a generic `sensor`, but for some things e.g. `binary_sensor` might be better; likewise some things could use a specified `device_class`.
